### ISSUE TITLE
System.Data.SqlClient.Tests crash on UapAot in Release-x86

### ISSUE THREAD

**Issue by safern** [dotnet/corefx#22416]
This seems to be a codegen bug as this only happens when ILC built in ret mode, if we build them as chk mode then they don't crash. 

There are various tests crashing and they seem to be flaky, once every 3 runs aprox.

This is one of the tests I've seen crash:
- System.Data.SqlClient.Tests.SqlErrorCollectionTest.CopyTo_Throws

Error:
```
Unhandled exception at 0x0F4456D1 (mrt100_app.dll) in xunit.console.netcore.exe: 0xC0000005: Access violation reading location 0x000003E8.
```

Stack Trace:
```
>	xunit.console.netcore.dll!$18_System::Data::SqlClient::SessionData..ctor() Line 72	Unknown
 	xunit.console.netcore.dll!$18_System::Data::SqlClient::SqlInternalConnectionTds..ctor($18_System::Data::ProviderBase::DbConnectionPoolIdentity & identity, $18_System::Data::SqlClient::SqlConnectionString & connectionOptions, System::Object & providerInfo, bool redirectedUserInstance, $18_System::Data::SqlClient::SqlConnectionString & reconnectSessionData, $18_System::Data::SqlClient::SessionData & applyTransientFaultHandling, bool) Line 323	Unknown
 	xunit.console.netcore.dll!$18_System::Data::SqlClient::SqlConnectionFactory.CreateConnection($18_System::Data::Common::DbConnectionOptions & options, $18_System::Data::Common::DbConnectionPoolKey & poolKey, System::Object & poolGroupProviderInfo, $18_System::Data::ProviderBase::DbConnectionPool & pool, $17_System::Data::Common::DbConnection & owningConnection, $18_System::Data::Common::DbConnectionOptions & userOptions) Line 136	Unknown
 	xunit.console.netcore.dll!$18_System::Data::ProviderBase::DbConnectionFactory.CreateNonPooledConnection($17_System::Data::Common::DbConnection & owningConnection, $18_System::Data::ProviderBase::DbConnectionPoolGroup & poolGroup, $18_System::Data::Common::DbConnectionOptions & userOptions) Line 100	Unknown
 	xunit.console.netcore.dll!$18_System::Data::ProviderBase::DbConnectionFactory.TryGetConnection($17_System::Data::Common::DbConnection & owningConnection, System::Threading::Tasks::TaskCompletionSource$1<$18_System::Data::ProviderBase::DbConnectionInternal> & retry, $18_System::Data::Common::DbConnectionOptions & userOptions, $18_System::Data::ProviderBase::DbConnectionInternal & oldConnection, $18_System::Data::ProviderBase::DbConnectionInternal & & connection) Line 278	Unknown
 	xunit.console.netcore.dll!$18_System::Data::ProviderBase::DbConnectionInternal.TryOpenConnectionInternal($17_System::Data::Common::DbConnection & outerConnection, $18_System::Data::ProviderBase::DbConnectionFactory & connectionFactory, System::Threading::Tasks::TaskCompletionSource$1<$18_System::Data::ProviderBase::DbConnectionInternal> & retry, $18_System::Data::Common::DbConnectionOptions & userOptions) Line 691	Unknown
 	xunit.console.netcore.dll!$18_System::Data::ProviderBase::DbConnectionClosed.TryOpenConnection($17_System::Data::Common::DbConnection & outerConnection, $18_System::Data::ProviderBase::DbConnectionFactory & connectionFactory, System::Threading::Tasks::TaskCompletionSource$1<$18_System::Data::ProviderBase::DbConnectionInternal> & retry, $18_System::Data::Common::DbConnectionOptions & userOptions) Line 1902	Unknown
 	xunit.console.netcore.dll!$18_System::Data::SqlClient::SqlConnection.TryOpen(System::Threading::Tasks::TaskCompletionSource$1<$18_System::Data::ProviderBase::DbConnectionInternal> & retry) Line 1031	Unknown
 	xunit.console.netcore.dll!$18_System::Data::SqlClient::SqlConnection.Open() Line 598	Unknown
 	xunit.console.netcore.dll!$19_System::Data::SqlClient::Tests::SqlErrorCollectionTest.CreateCollection() Line 144	Unknown
 	xunit.console.netcore.dll!$19_System::Data::SqlClient::Tests::SqlErrorCollectionTest.ValidateCopyToThrows(System::Action$3<$18_System::Data::SqlClient::SqlErrorCollection,$18_System::Data::SqlClient::SqlError[],System::Int32> & copyTo, System::Action$1<$18_System::Data::SqlClient::SqlErrorCollection> & additionalValidation) Line 93	Unknown
 	xunit.console.netcore.dll!$19_System::Data::SqlClient::Tests::SqlErrorCollectionTest.CopyTo_Throws() Line 75	Unknown
```

I will force the tests to be built as chk for ILC towards getting a clean build in the .csproj adding a comment linked to this issue so that whenever this issue is solved we remove that line I will add.

cc: @joshfree

**Comment by joshfree** [comment]
@hoyMS could you take a look?

**Comment by hoyMS** [comment]
sure, someone from my team is taking a look.

**Comment by hoyMS** [comment]
A fix has been checked into TFS as 1667773.
